crates/s3s/src/validation.rs

```diff
@@ -0,0 +1,64 @@
+//! Validation API for S3 bucket names
+
+/// Trait for validating S3 names
+///
+/// Implementations should return `true` for valid names and `false` for invalid ones.
+pub trait NameValidation: Send + Sync {
+    /// Validate a bucket name
+    fn validate_bucket_name(&self, name: &str) -> bool;
+}
+
+/// Default AWS-compliant name validation
+#[derive(Debug, Clone, Default)]
+pub struct DefaultNameValidation;
```

Comment:
We may reserve the constructors of built-in `NameValidation` types in case we add some options in the future. `AwsNameValidation` with `Default` trait is ok.